stdlib/public/core/Collection.swift

```diff
@@ -661,6 +664,7 @@ public protocol MutableCollection : MutableIndexable, Collection {
   /// within an algorithm, but when that fails, invoking the
   /// same algorithm on `body`\ 's argument lets you trade safety for
   /// speed.
+  @discardableResult
   mutating func _withUnsafeMutableBufferPointerIfSupported<R>(
```

Comment:
Please drop this one, for the same reason as you explained on `withUnsafeMutableBufferPointer`.